Example of a Hydra job that is failing because of this issue: https://hydra.nixos.org/build/242943226
Hydra log: https://hydra.nixos.org/build/242943226/nixlog/4

Relevant log excerpt:

```
-- Build files have been written to: /build/ustream-ssl-498f6e2/build
cmake: enabled parallel building
cmake: enabled parallel installing
Running phase: buildPhase
build flags: -j2 SHELL=/nix/store/l34lhs5k16k8pydj2rjq70pqxz86kd3f-bash-5.2-p21/bin/bash
[ 25%] Building C object CMakeFiles/ustream-ssl.dir/ustream-io-openssl.c.o
[ 25%] Building C object CMakeFiles/ustream-ssl.dir/ustream-ssl.c.o
In file included from /build/ustream-ssl-498f6e2/ustream-io-openssl.c:24:
/build/ustream-ssl-498f6e2/openssl_bio_compat.h:4:10: fatal error: openssl/opensslv.h: No such file or directory
    4 | #include <openssl/opensslv.h>
      |          ^~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [CMakeFiles/ustream-ssl.dir/build.make:90: CMakeFiles/ustream-ssl.dir/ustream-io-openssl.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
In file included from /build/ustream-ssl-498f6e2/ustream-internal.h:27,
                 from /build/ustream-ssl-498f6e2/ustream-ssl.c:25:
/build/ustream-ssl-498f6e2/ustream-openssl.h:26:10: fatal error: openssl/ssl.h: No such file or directory
   26 | #include <openssl/ssl.h>
      |          ^~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [CMakeFiles/ustream-ssl.dir/build.make:76: CMakeFiles/ustream-ssl.dir/ustream-ssl.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:87: CMakeFiles/ustream-ssl.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

--- a/pkgs/development/libraries/ustream-ssl/default.nix
+++ b/pkgs/development/libraries/ustream-ssl/default.nix
@@ -1,4 +1,12 @@
-{ stdenv, lib, fetchgit, cmake, pkg-config, libubox-nossl, ssl_implementation }:
+{ stdenv
+, lib
+, fetchgit
+, cmake
+, pkg-config
+, libubox-nossl
+, ssl_implementation
+, additional_buildInputs ? [ ]
+}:
 
 stdenv.mkDerivation {
   pname = "ustream-ssl";
@@ -21,7 +29,7 @@ stdenv.mkDerivation {
   cmakeFlags = [ "-D${lib.toUpper ssl_implementation.pname}=ON" ];
 
   nativeBuildInputs = [ cmake pkg-config ];
-  buildInputs = [ ssl_implementation ];
+  buildInputs = [ ssl_implementation ] ++ additional_buildInputs;
 
   passthru = {
     inherit ssl_implementation;

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -20955,7 +20955,7 @@ with pkgs;
 
   ustream-ssl = callPackage ../development/libraries/ustream-ssl { ssl_implementation = openssl; };
 
-  ustream-ssl-wolfssl = callPackage ../development/libraries/ustream-ssl { ssl_implementation = wolfssl; };
+  ustream-ssl-wolfssl = callPackage ../development/libraries/ustream-ssl { ssl_implementation = wolfssl; additional_buildInputs = [ openssl ]; };
 
   ustream-ssl-mbedtls = callPackage ../development/libraries/ustream-ssl { ssl_implementation = mbedtls_2; };
 

